crowbar_framework/app/models/service_object.rb

```diff
@@ -1084,6 +1084,17 @@ def apply_role(role, inst, in_queue)
     element_order = new_deployment["element_order"]
 
     #
+    # stop chef daemon on all nodes
+    #
+    nodes_to_freeze_chef_service = new_elements.values.flatten
+    nodes_to_freeze_chef_service.each do |node_name|
+      node = NodeObject.find_node_by_name(node_name)
+      @logger.debug "stopping chef service on #{node_name} before apply_role"
+      node.service :chef, :stop
+      wait_for_chef_clients(node_name)
+    end
```

Comment:
A few things:
* this is too early (a few lines below, we might put the apply in the queue)
* this is not taking clusters in consideration (you have `expanded_new_elements` for that a few lines later)